farmfs/snapshot.py

```diff
@@ -93,11 +105,12 @@ def key_snap_iterator():
       for item in self.data:
         if isinstance(item, list):
           assert len(item) == 3
-          yield SnapshotItem(*item, reverser=self._reverser)
+          parsed = SnapshotItem(*item, reverser=self._reverser)
```

Comment:
We've embedded the KeySnapshot sins into SnapshotItem! Lets move the logic for parsing snapshots back out here.